File: app/src/main/java/com/example/mobil_programlama_proje/database/entity/Note.java
package com.example.mobil_programlama_proje.database.entity;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity(
        foreignKeys = {
                @ForeignKey(entity = User.class,
                        parentColumns = "userId",
                        childColumns = "userOwnerId"),

                @ForeignKey(entity = Category.class,
                        parentColumns = "categoryId",
                        childColumns = "categoryOwnerId")
        }
)
public class Note {
    @PrimaryKey(autoGenerate = true)
    public int noteId;

    public String title;
    public String content;

    public int userOwnerId;
    public int categoryOwnerId;
}

File: app/src/main/java/com/example/mobil_programlama_proje/model/Note.kt
package com.example.mobil_programlama_proje.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.UUID

@Entity(tableName = "notes")
data class Note(
    @PrimaryKey
    val id: String = UUID.randomUUID().toString(),

    val title: String,
    val content: String,

    val userId: Int? = null,

    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
)